recipes/dcmtk/all/patches/0002-cmake-use-conan-packages.patch

```diff
@@ -0,0 +1,219 @@
+@@ -310,8 +312,8 @@ else()
+   if(DCMTK_WITH_ZLIB)
+     if(WITH_ZLIBINC)
+       set(ZLIB_INCDIR "${WITH_ZLIBINC}/include")
+-      set(ZLIB_LIBDIR "${WITH_ZLIBINC}/lib")
+-      set(ZLIB_LIBS debug "${ZLIB_LIBDIR}/zlib_d.lib" optimized "${ZLIB_LIBDIR}/zlib_o.lib")
++      link_directories(${WITH_ZLIBINC}/lib)
++      set(ZLIB_LIBS CONAN_PKG::zlib)
+       message(STATUS "Info: DCMTK ZLIB support will be enabled")
+       set(WITH_ZLIB 1)
+     else() # turn off library if library path not set
+@@ -325,8 +327,8 @@ else()
+   if(DCMTK_WITH_SNDFILE)
+     if(WITH_SNDFILEINC)
+       set(SNDFILE_INCDIR "${WITH_SNDFILEINC}/include")
+-      set(SNDFILE_LIBDIR "${WITH_SNDFILEINC}/lib")
+-      set(SNDFILE_LIBS debug "${SNDFILE_LIBDIR}/libsndfile_d.lib" optimized "${SNDFILE_LIBDIR}/libsndfile_o.lib")
++      link_directories(${WITH_SNDFILEINC}/lib)
++      set(SNDFILE_LIBS CONAN_PKG::libsndfile)
+       message(STATUS "Info: DCMTK SNDFILE support will be enabled")
+       set(WITH_SNDFILE 1)
+     else() # turn off library if library path not set
+@@ -340,8 +342,8 @@ else()
+   if(DCMTK_WITH_ICONV)
+     if(WITH_LIBICONVINC)
+       set(LIBICONV_INCDIR "${WITH_LIBICONVINC}/include")
+-      set(LIBICONV_LIBDIR "${WITH_LIBICONVINC}/lib")
+-      set(LIBICONV_LIBS debug "${LIBICONV_LIBDIR}/libiconv_d.lib" optimized "${LIBICONV_LIBDIR}/libiconv_o.lib")
++      link_directories(${WITH_LIBICONVINC}/lib)
++      set(LIBICONV_LIBS CONAN_PKG::libiconv)
+       message(STATUS "Info: DCMTK ICONV support will be enabled")
+       set(WITH_LIBICONV 1)
+     else() # turn off library if library path not set
+@@ -363,8 +365,8 @@ else()
+           set(WITH_OPENJPEG "")
+       else()
+           set(OPENJPEG_INCDIR "${WITH_OPENJPEGINC1}")
+           set(OPENJPEG_LIBDIR "${WITH_OPENJPEGINC}/lib")
+-          set(OPENJPEG_LIBS debug "${OPENJPEG_LIBDIR}/openjp2_d.lib" optimized "${OPENJPEG_LIBDIR}/openjp2_o.lib")
++          set(OPENJPEG_LIBS CONAN_PKG::openjpeg)
+           message(STATUS "Info: DCMTK OpenJPEG support will be enabled")
+           set(WITH_OPENJPEG 1)
+       endif()
+@@ -388,7 +390,7 @@ if(NOT DEFINED DCMTK_WITH_STDLIBC_ICONV)
+ endif()
+ 
+ if(DCMTK_WITH_ICU)
+-  find_package(ICU COMPONENTS uc data QUIET)
++  find_package(ICU COMPONENTS uc data REQUIRED)
+   if(NOT ICU_FOUND)
+     message(STATUS "Warning: ICU support will be disabled because the ICU were not found.")
+     set(DCMTK_WITH_ICU OFF CACHE BOOL "" FORCE)
+@@ -404,7 +406,7 @@ endif()
+ 
+ # Find doxygen
+ if(DCMTK_WITH_DOXYGEN)
+-  find_package(Doxygen QUIET) # will set variable DOXYGEN_EXECUTABLE
++  find_package(Doxygen REQUIRED) # will set variable DOXYGEN_EXECUTABLE
+   if(NOT DOXYGEN_FOUND)
+     message(STATUS "Warning: DOXYGEN support will be disabled because doxygen was not found.")
+     set(DCMTK_WITH_DOXYGEN OFF CACHE BOOL "" FORCE)
```

Comment:
doxygen is not used by this recipe, I think.
```suggestion
```

Reply:
I wanted to keep just the one existing patch used for 3.6.5 plus the libxml modification but git will not cleanly apply it on 3.6.6 because it gets confused by an `if()/else()/endif()` change early on in the area targeted by the patch, so I just recreated the same changes in the second patch so they apply cleanly on 3.6.6.   You are correct Doxygen is always set off by the recipe. The existing patch has changed all `find_package(... QUIET` to `find_package(... REQUIRED` regardless of whether the recipe might enable them, e.g. also for openjpeg and sndfile.  I think it is harmless to leave it this way and one less thing to find and change if the recipe later wants to add options to enable them?  At least the two patches should be consistent.

Reply:
It's harmless indeed.

Reply:
Resolving per discussion.